pkg/operator/operator.go

```diff
@@ -80,35 +78,39 @@ func NewOperator() (context.Context, *Operator) {
 	ctx = knativeinjection.WithNamespaceScope(ctx, system.Namespace())
 
 	// Options
-	opts := lo.Must(options.New().Parse(os.Args[1:]...))
+	ctx = injection.WithOptionsOrDie(ctx, options.Injectables...)
 
 	// Make the Karpenter binary aware of the container memory limit
 	// https://pkg.go.dev/runtime/debug#SetMemoryLimit
-	if opts.MemoryLimit > 0 {
-		newLimit := int64(float64(opts.MemoryLimit) * 0.9)
+	if options.FromContext(ctx).MemoryLimit > 0 {
+		newLimit := int64(float64(options.FromContext(ctx).MemoryLimit) * 0.9)
 		debug.SetMemoryLimit(newLimit)
 	}
 
 	// Webhook
 	ctx = webhook.WithOptions(ctx, webhook.Options{
-		Port:        opts.WebhookPort,
-		ServiceName: opts.ServiceName,
-		SecretName:  fmt.Sprintf("%s-cert", opts.ServiceName),
+		Port:        options.FromContext(ctx).WebhookPort,
+		ServiceName: options.FromContext(ctx).ServiceName,
+		SecretName:  fmt.Sprintf("%s-cert", options.FromContext(ctx).ServiceName),
 		GracePeriod: 5 * time.Second,
 	})
 
 	// Client Config
 	config := controllerruntime.GetConfigOrDie()
-	config.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(float32(opts.KubeClientQPS), opts.KubeClientBurst)
+	config.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(float32(options.FromContext(ctx).KubeClientQPS), options.FromContext(ctx).KubeClientBurst)
 	config.UserAgent = appName
 
 	// Client
 	kubernetesInterface := kubernetes.NewForConfigOrDie(config)
 
 	// Inject settings from the ConfigMap(s) into the context
 	ctx = injection.WithSettingsOrDie(ctx, kubernetesInterface, apis.Settings...)
-	opts = opts.MergeSettings(settings.FromContext(ctx))
-	ctx = options.ToContext(ctx, opts)
+
+	// Temporarily merge settings into options until configmap is removed
+	// Note: injectables are pointer to those already in context
+	for _, o := range options.Injectables {
+		o.MergeSettings(ctx)
```

Comment:
This doesn't look like it updates the context, should the interface show that it updates the context by outputting a new context. I get that this is a pointer but not returning a new context doesn't make it super clear that something is being updated here

Reply:
Ya I'm a bit mixed on that, that's why I left the comment on line 110. Maybe that can be more clear. I'm hesitant to make it return a context because I think it just changes the location where something isn't clear, since it wouldn't be clear why the function returns a context in the first place. That's why I felt a comment was more appropriate. Or is that pattern just common enough in the go / k8s world that its clear regardless?

Reply:
Resolved offline.